src/plugins/ui_feature/find_and_replace.ts

```diff
@@ -222,47 +328,71 @@ export class FindAndReplacePlugin extends UIPlugin {
     }
     const matches = this.searchMatches;
     const selectedMatch = matches[this.selectedMatchIndex];
-    const sheetId = this.getters.getActiveSheetId();
-    const cell = this.getters.getCell({ sheetId, ...selectedMatch });
+    const cell = this.getters.getCell(selectedMatch);
     if (cell?.isFormula && !this.searchOptions.searchFormulas) {
-      this.selectNextCell(Direction.next);
-    } else {
-      const replaceRegex = new RegExp(
-        this.currentSearchRegex.source,
-        this.currentSearchRegex.flags + "g"
-      );
-      const toReplace: string | null = this.getSearchableString({
-        sheetId,
-        col: selectedMatch.col,
-        row: selectedMatch.row,
-      });
-      const newContent = toReplace.replace(replaceRegex, replaceWith);
-      this.dispatch("UPDATE_CELL", {
-        sheetId: this.getters.getActiveSheetId(),
-        col: selectedMatch.col,
-        row: selectedMatch.row,
-        content: newContent,
-      });
-      this.searchMatches.splice(this.selectedMatchIndex, 1);
-      this.selectNextCell(Direction.current);
+      return this.selectNextCell(Direction.next);
+    }
+    const replaceRegex = new RegExp(
+      this.currentSearchRegex.source,
+      this.currentSearchRegex.flags + "g"
+    );
```

Comment:
We recently bumped our version of JS, we can drop this part and use `replaceAll`

Reply:
when using replaceAll, it's important to consider the case sensitivity. Here's the scenario:

1. We've got a cell with the content "FORMAT."
2. Now, let's say we want to find the word "for" and replace it with "dor."

Due to its case sensitivity, replaceAll won't allow us to make that change.

Reply:
when using replaceAll, it's important to consider the case sensitivity. Here's the scenario:

1. We've got a cell with the content "FORMAT."
2. Now, let's say we want to find the word "for" and replace it with "dor."

Due to its case sensitivity, replaceAll won't allow us to make that change.

Reply:
I don't see why the case sensitivity change anything. The flag for case sensitivity will be in `this.currentSearchRegex`, calling replace or replaceAll don't change that ?

Reply:
[ReplaceAll](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) take regex as paremeter, you don't have to  rebuild it with the global `g` anymore

Reply:
If I don't rebuild regex with global `g` I'm getting [error](https://deepscan.io/docs/rules/bad-replace-all-arg). This is also written in above mentioned documentation. I may be wrong but I'm a bit confused since both things are contradicting.

Reply:
I tried doing that, but it's still effecting the test cases. Whenever we search for =1 in test case, it won't work but in UI it seems to be working fine. Unless I'm doing something wrong. One more thing, these failing test cases are getting error on search regex but our fix is implemented for replace. Do you have any other approach in mind?